app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/react/DeployStatusPills.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { IManifest } from '@spinnaker/core';
+
+interface IStatusClasses {
+  [key: string]: string;
+}
+
+const STATUS_PILLS: IStatusClasses = {
+  available: 'success',
+  stable: 'success',
+  paused: 'warn',
+  failed: 'danger',
+};
+
+interface IDeployManifestStatusProps {
+  manifest: IManifest;
+}
+
+export class DeployManifestStatusPills extends React.Component<IDeployManifestStatusProps> {
+  public render() {
+    const { manifest } = this.props;
+    if (manifest == null || manifest.status == null) {
+      return null;
+    }
+    return Object.keys(manifest.status).map((statusKey, i) => {
+      const statusDescription = manifest.status[statusKey];
+      const statusClass = STATUS_PILLS[statusKey] || '';
```

Comment:
Would defaulting this to `warn` make the logic below less complex?  Or perhaps pulling some of the logic in the template into a variable, `const showStatus: bool = statusDescription.state || statusDescription.message`?

Reply:
I think I'm confused about the difference between the `statusKey` and the `statusDescription.state`...what's the intended use of each of these fields (and why are we not using the state at all here)?

Reply:
`statusDescription.state` is a boolean that indicates, I think, stability of the status.  `statusKey` is the string name of the status.  So if the state is stable we show the colour pill that matches the statusKey.  If the state is unstable but there is a message we show a warning pill and the message will be shown on hover.

Reply:
i've updated this to name the booleans and make logic clearer.

Reply:
Thanks for the explanation/change---this makes sense to me now.